File: app/src/main/java/com/cna/mineru/cna/Adapter/GridAdapter.java
package com.cna.mineru.cna.Adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.cna.mineru.cna.DB.ExamSQLClass;
import com.cna.mineru.cna.DTO.HomeData;
import com.cna.mineru.cna.R;

import java.util.ArrayList;

/*
    HomeFragment의 각 Item 요소 표현을 위한 Adapter
    HomeFragment

 */

public class GridAdapter extends BaseAdapter {
    Context context;
    int layout;
    LayoutInflater inf;
    ArrayList<HomeData> items;
    public GridAdapter(Context context, int layout, ArrayList<HomeData> list) {
        items = new ArrayList();
        this.context = context;
        this.layout = layout;
        items=list;
        inf = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    public void addItem(HomeData item){
        items.add(0,item);
    }
    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public Object getItem(int position) {
        return items.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @SuppressLint("SetTextI18n")
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView==null)
            convertView = inf.inflate(layout, null);
        TextView tv = (TextView) convertView.findViewById(R.id.titleText);
        RelativeLayout btn_count = (RelativeLayout) convertView.findViewById(R.id.btn_count);
        ImageView iv_count = (ImageView) convertView.findViewById(R.id.iv_count);
        TextView tv_count = (TextView) convertView.findViewById(R.id.tv_count);
        HomeData item = items.get(position);
        tv.setText(item.title_text);
//        tv_count.setText(""+db.getNoteCount(position+1));
        tv_count.setText("");
        return convertView;
    }
}

File: app/src/main/java/com/cna/mineru/cna/Adapter/GridAdapter.kt
package com.cna.mineru.cna.Adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView

import com.cna.mineru.cna.DTO.HomeData
import com.cna.mineru.cna.R

import java.util.ArrayList

/*
    HomeFragment의 각 Item 요소 표현을 위한 Adapter
    HomeFragment

 */

class GridAdapter(
    internal var context: Context?,
    internal var layout: Int?,
    list: ArrayList<HomeData>?
) : BaseAdapter() {
    internal var inf: LayoutInflater
    internal var items: ArrayList<HomeData> = ArrayList()

    init {
        if (list != null) {
            items = list
        }
        inf = context?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    fun addItem(item: HomeData) {
        items.add(0, item)
    }

    override fun getCount(): Int {
        return items.size
    }

    override fun getItem(position: Int): Any {
        return items[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    @SuppressLint("SetTextI18n")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null)
            convertView = layout?.let { inf.inflate(it, null) }
        val tv = convertView!!.findViewById<View>(R.id.titleText) as TextView
        val btn_count = convertView.findViewById<View>(R.id.btn_count) as RelativeLayout
        val iv_count = convertView.findViewById<View>(R.id.iv_count) as ImageView
        val tv_count = convertView.findViewById<View>(R.id.tv_count) as TextView
        val item = items[position]
        tv.setText(item.title_text)
        //        tv_count.setText(""+db.getNoteCount(position+1));
        tv_count.text = ""
        return convertView
    }
}
